myokit/formats/sbml/_parser.py

```diff
@@ -0,0 +1,1206 @@
+        """
+        # Get composing base units
+        units = unit_def.findall(
+            './'
+            + '{http://www.sbml.org/sbml/level3/version2/core}'
+            'listOfUnits/'
+            + '{http://www.sbml.org/sbml/level3/version2/core}'
+            + 'unit')
+        if not units:
+            return None
+
+        # instantiate unit definition
+        unit_def = myokit.units.dimensionless
+
+        # construct unit definition from base units
+        for baseUnit in units:
+            kind = baseUnit.get('kind')
+            if not kind:
+                raise SBMLError(
+                    'The file does not adhere to SBML 3.2 standards.'
+                    ' No unit kind provided.')
+            myokitUnit = sbml_to_myokit_unit_dict[kind]
+            myokitUnit *= float(baseUnit.get('multiplier', default=1.0))
+            myokitUnit *= 10 ** float(baseUnit.get('scale', default=0.0))
+            myokitUnit **= float(baseUnit.get('exponent', default=1.0))
+
+            # "add" composite unit to unit definition
+            unit_def *= myokitUnit
+
+        return unit_def
+
+    def _get_units(self, parameter, user_unit_dict):
+        """
+        Returns :class:myokit.Unit expression of the unit of a parameter.
+        """
+        unit = parameter.get('units')
+        if unit in user_unit_dict:
+            return user_unit_dict[unit]
+        elif unit in sbml_to_myokit_unit_dict:
+            return sbml_to_myokit_unit_dict[unit]
+        else:
+            return None
+
+    def _get_substance_units(self, species, user_unit_dict):
+        """
+        Returns :class:myokit.Unit expression of the unit of a species.
+        """
+        # Convert substance unit into myokiy.Unit
+        unit = species.get('substanceUnits')
+        if unit in user_unit_dict:
+            return user_unit_dict[unit]
+        elif unit in sbml_to_myokit_unit_dict:
+            return sbml_to_myokit_unit_dict[unit]
+        else:
+            return None
+
+    def _get_species_initial_value_in_amount(self, species, compId):
+        """
+        Returns the initial value of a species either in amount or
+        concentration depend on the flag is Amount.
+        """
+        amount = species.get('initialAmount')
+        if amount:
+            return amount
+        conc = species.get('initialConcentration')
+        if conc:
+            volume = param_and_species_dict[compId]
+            return myokit.Multiply(
+                myokit.Number(amount), myokit.Name(volume))
+        return None
+
+
+class SBMLError(myokit.ImportError):
+    """
+    Thrown if an error occurs when importing SBML
+    """
+
+
+# SBML base units according to libSBML
+sbml_to_myokit_unit_dict = {
+    'ampere': myokit.units.A,
+    'avogadro': myokit.units.mol,  # This is a simplification of SBML's def
+    'becquerel': myokit.units.Bq,
+    'candela': myokit.units.cd,
+    'celsius': myokit.units.C,
```

Comment:
This one's not right. Myokit doesn't support celsius (or fahrenheit). So you'd have to catch this somewhere else and raise an exception if someone tries to use celsius

Reply:
ok I wrapped the `smbl_to_myokit_dict` in a function that throws an error when celsius is passed